src/main/webapp/script.js

```diff
@@ -41,47 +42,71 @@ function initCommentSlider() {
  * @return {JSON} comments json object of comment objects
  */
 async function getYouTubeComments(url) {
-  url = extractYouTubeUrl(url);
-  const numComments = document.getElementById(SLIDER_NAME).value;
-  const response = await fetch('/YouTubeComments?url=' + url +
-    '&numComments=' + numComments);
-  const comments = await response.json();
-  return comments;
+  try {
+    url = extractYouTubeUrl(url);
+    const numComments = document.getElementById(SLIDER_NAME).value;
+    const response = await fetch('/YouTubeComments?url=' + url +
+      '&numComments=' + numComments);
+    const comments = await response.json();
+    return comments;
+  } catch (err) {
+    throw new Error('Error retrieving comments.');
+  }
 }
 
 /**
  * Wrapper function for preparing onClick function
  */
 function onButtonPress() {
   $('#submit-link-btn').click(function() {
-    showLoadingGif();  
-    const urlInput = document.getElementById('link-input').value;
-    updateUIWithVideoContext(urlInput);
-    displayOverallResults(urlInput);
+    try {
+      toggleErrorOutput('none');
+      showLoadingGif();
+      const urlInput = document.getElementById('link-input').value;
+      updateUIWithVideoContext(urlInput);
+      displayOverallResults(urlInput);
+    } catch (err) {
+      displayError(err.message);
+    }
   });
 }
 
+// Toggles visibility of error message
+function toggleErrorOutput(mode) {
+  document.getElementById(ERROR_OUTPUT_ID).style.display = mode;
+}
+
+function displayError(message) {
+  hideLoadingGif();
+  toggleErrorOutput('block');
+  document.getElementById('error-details').innerText = message;
+}
+
 /**
  * Fetches data and adds to html
  * @param {String} url youtube url to retrieve comments
  */
 async function displayOverallResults(url) {
+  try {
```

Comment:
do try catch only for getYoutubeComemnts. There are other things like clearElement that should be done at whether there is error or not

Reply:
All the frontend functions have try/catch blocks to give details on what went wrong.

Reply:
clearElement() is purely a frontend method that has no backend dependency, we don't need to do a try catch for it. 

When to use try-catch statement?
"The best time to use try-catch is in portions of your code where you suspect errors will occur that are beyond your control, for whatever reasons."
https://www.impressivewebs.com/javascript-try-catch/

Reply:
My bad, I think I misunderstood what you were saying, I agree now that I should move the try/catch boundaries